### PR TITLE
[ios] Implement macOS trackpad and mouse events (scroll, rotate, zoom) support

### PR DESCRIPTION
**Pull request by kirylkaveryn**
Closes https://github.com/organicmaps/organicmaps/issues/2524

This PR implements:
1. mouse zoom
2. trackpad 2-finger pinch to zoom
3. trackpad 2-finger rotation

https://github.com/user-attachments/assets/cf90df5c-93fa-45c3-b32b-8ce5ee975cad

### CODE REVIEW

**Review by vng — COMMENTED** [review]
Well, not a big deal here, but these static vars add _fly in the honey_ :)
static CGFloat currentScale
static CGFloat currentRotation

Can we save the state at least in MapViewController instance?

File: iphone/Maps/Classes/MapViewController.mm [hunk 1]
```diff
@@ -755,4 +787,85 @@ - (void)goBack {
...
+  switch (recognizer.state) {
+    case UIGestureRecognizerStateBegan:
+    case UIGestureRecognizerStateChanged:
+    {
+      self.userTouchesAction = UserTouchesActionDrag;
+      GetFramework().Rotate(currentRotation == 0 ? recognizer.rotation : currentRotation + recognizer.rotation, false);
+      break;
+    }
+    case UIGestureRecognizerStateEnded:
+      currentRotation += recognizer.rotation;
```
[the remark this answers is not in the document]
Comment (kirylkaveryn) [reply iphone/Maps/Classes/MapViewController.mm:849]: Rotation here is related to the current angle and starts from the 0 on every recognized gesture.
So if the first rotation gesture is:
0, 1.0, 5.0... end of gesture (the user removes the fingers) 
The next rotation starts from zero too.

The core framework uses the **absolute rotation angle** but **related** to the previous state **scale**, which is why these values are calculated differently.

**Comment by kirylkaveryn** [comment]
> Well, not a big deal here, but these static vars add _fly in the honey_ :) static CGFloat currentScale static CGFloat currentRotation
> 
> Can we save the state at least in MapViewController instance?

Fixed! Thanks!

**Review by vng — APPROVED** [review]

**Comment by samcarter** [comment]
Thank you so much! The mouse zoom makes organic macs so much more convenient to use!

In case anybody is having trouble: check if in the app settings (`cmd+,`), the option `Touch Alternatives` is disabled, otherwise the new mouse zoom won't work.

**Pull request merged**